src/providers/ldap/sdap.c

```diff
@@ -648,10 +648,13 @@ static bool objectclass_matched(struct sdap_attr_map *map,
         len = strlen(objcl) + 1;
     }
 
-    if (strncasecmp(map[SDAP_OC_GROUP].name, objcl, len) == 0) {
+    if (strncasecmp(map[0].name, objcl, len) == 0) {
         return true;
     }
 
+    /* BUG, Fixme: we use objectclass_matched() for multiple maps, not only groups
+     * so this code might popentially yield into unexpected results */
```

Comment:
Hi,

good catch, I think this can be fixed by adding `strcmp(map[SDAP_OC_GROUP_ALT].sys_name, SYSDB_GROUP_CLASS)`

bye,
Sumit

Reply:
Hi,
Like below?

Reply:
Hi,

yes, but I would keep `SDAP_OC_GROUP_ALT` because this if-block is explicitly for this. A comment would be better at the if-block before saying that the first (`0`) entry in the maps is always the object class.

bye,
Sumit

Reply:
you mean only the last if block right?

Reply:
> you mean only the last if block right?

Yes, with respect to keep `SDAP_OC_GROUP_ALT`.

Reply:
Thanks.